File: Cc.Core/Models/Document.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Document
    {
        public Document()
        {
            Principal = new HashSet<Principal>();
        }

        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string? InsertUser { get; set; }
        public DateTime UpdateDate { get; set; }
        public string? UpdateUser { get; set; }
        public string? Status { get; set; }
        public string? StatusMessage { get; set; }
        public string? Entity { get; set; }
        public string? Type { get; set; }
        public string? Name { get; set; }
        public int? Date { get; set; }
        public byte[]? File { get; set; }
        public string? Repository { get; set; }
        public string? ExternalId { get; set; }
        public byte[]? FileSigned { get; set; }
        public string? RepositorySigned { get; set; }
        public string? ExternalIdSigned { get; set; }
        public bool Opened { get; set; }

        public virtual ICollection<Principal> Principal { get; set; }
    }
}

File: Cc.Ui/src/models/document.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('document')
export class Document extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    status: string;
    @field()
    statusMessage: string;
    @field()
    entity: string;
    @field()
    type: string;
    @field()
    name: string;
    @field()
    file: ArrayBuffer;
    @field()
    repository: string;
    @field()
    externalId: string;
    @field()
    fileSigned: ArrayBuffer;
    @field()
    repositorySigned: string;
    @field()
    externalIdSigned: string;
    @field()
    opened: boolean;
    @field()
    date: string;
    //Extra field for upload
    fileEncoded: string

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Document {

        let item = new Document();
        return item;
    }
}

@resource('document')
export class DocumentSearchModel extends Entity implements Pagination {
    
    @field()
    status?: string;
    @field()
    statusMessage?: string;
    @field()
    entity?: string;
    @field()
    type?: string;
    @field()
    name?: string;
    @field()
    file?: ArrayBuffer;
    @field()
    repository?: string;
    @field()
    externalId?: string;
    @field()
    fileSigned?: ArrayBuffer;
    @field()
    repositorySigned?: string;
    @field()
    externalIdSigned?: string;
    @field()
    opened?: boolean;
    @field()
    date?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}